gadgets/snapshot_socket/program.bpf.c

```diff
@@ -50,13 +50,48 @@
 struct socket_entry {
 	struct gadget_l4endpoint_t src;
 	struct gadget_l4endpoint_t dst;
+	char state[16];
 	__u32 state;
 	__u32 ino;
 	gadget_netns_id netns_id;
 };
 
 GADGET_SNAPSHOTTER(sockets, socket_entry, ig_snap_tcp, ig_snap_udp);
 
+typedef enum  {
+    STATE_UNKNOWN = 0,
+    STATE_ESTABLISHED = 1,
+    STATE_SYN_SENT = 2,
+    STATE_SYN_RECV = 3,
+    STATE_FIN_WAIT1 = 4,
+    STATE_FIN_WAIT2 = 5,
+    STATE_TIME_WAIT = 6,
+    STATE_CLOSE = 7,
+    STATE_CLOSE_WAIT = 8,
+    STATE_LAST_ACK = 9,
+    STATE_LISTEN = 10,
+    STATE_CLOSING = 11,
+} socket_state;
+
+
+static const char *state_to_string(__u32 state)
```

Comment:
You don't need to do this manually as Inspektor Gadget framework will do this for you. Please check [Enumerations documentation](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/gadget-devel/gadget-ebpf-api.md#enumerations).

However, I'm wondering how to consider the logic we had in the built-in gadget for `UDP` sockets:

https://github.com/inspektor-gadget/inspektor-gadget/blob/cf695bcf7fce116f8420436b2b874dfc97436703/pkg/gadgets/snapshot/socket/tracer/tracer.go#L63-L73

Maybe adding `STATE_ACTIVE` and `STATE_INACTIVE` to the enum and setting those two values based on the protocol?